### ISSUE TITLE
Projects not modular, duplicate packages across JARs

### ISSUE THREAD

**Issue by credmond** [square/wire#3204]
There are projects that "export" the same packages (com.squareup.wire).

E.g.: wire.runtime.jvm and wire.schema.jvm 

...which leads to errors like this, when building modular projects.
```
Error occurred during initialization of boot layer
java.lang.module.ResolutionException: Modules wire.runtime.jvm and wire.schema.jvm export package com.squareup.wire to module <something>
```

I don't know about the KT world, but this in a breach of JPMS guidelines.

Why can't we have modules-info.java or at least why do you have different JARs with the _same_ packages?

Or am I missing something altogether in how to use these libs...

**Comment by oldergod** [comment]
Assuming that JPMS is for Java Platform Module System, I fail to see what exporting means here.
Different modules can't have classes defined in the same package? If modules-info.java can fix it, that'd be great if you could push a PR.

**Comment by JakeWharton** [comment]
> Why can't we have modules-info.java

We can, but it involves doing separate compilation and creating a MR jar because our minimum is below 9.

> why do you have different JARs with the same packages?

Because effectively no one uses modules, this project predates modules existing, and no one has complained.

We can probably just move the schema type that offends. Binary compatibility of that module has been broken many times without a major version bump before.

If we do this, is the automatic module name still enough, or is there an actual reason to do module definitions?

**Comment by credmond** [comment]
> Assuming that JPMS is for Java Platform Module System, I fail to see what exporting means here. Different modules can't have classes defined in the same package? If modules-info.java can fix it, that'd be great if you could push a PR.

Yes, you cannot have two modules (i.e., JARs) with the same packages containing classes intended for external use, i.e., where the same package gets "exported" twice. 

modules-info.java cannot fix it alone, the classes would need to be moved in one of the two conflicting projects, assuming they're classes that you need exposed. I've noticed this sort of thing in quite a few square modules, in the past.
